src/bin/download_sysext.rs

```diff
@@ -262,7 +270,10 @@ async fn main() -> Result<(), Box<dyn Error>> {
     for pkg in pkgs_to_dl.iter_mut() {
         pkg.check_download(&unverified_dir)?;
 
-        pkg.download(&unverified_dir, &client).await?;
+        match pkg.download(&unverified_dir, &client).await {
+            Ok(_) => (),
+            _ => return Err(format!("unable to download \"{}\"", pkg.name).into()),
+        };
```

Comment:
matter of taste, but you could probably do this as something like

```rust
pkg.download(&unverified_dir, &client).await
    .map_err(|_| format!("unable to download \"{}\"", pkg.name).into())?;
```

Reply:
Changing that, I am getting this:
```
error[E0283]: type annotations needed
   --> src/bin/download_sysext.rs:274:22
    |
274 |             .map_err(|_| format!("unable to download \"{}\"", pkg.name).into())?;
    |                      ^^^                                                       - type must be known at this point
```

For now, I left that unchanged.

Reply:
In my other branch for anyhow migration, this kind of error mapping has been being already [done](https://github.com/flatcar/ue-rs/commit/f770af43d08e0a396266362a3acb5d1268c2ddf5#diff-42d8da6bd9394771306af4f16265896214311da363105863c47c2a9960a7eb82R79). 
Fortunately anyhow seems to allow that, without compile error.
Let's do the change in the anyhow migration PR, which I will create after merging the pending PRs. Ok?